KS-19-04-tank

DenisGubernatorov opened the pull request:
1) Энум с типами зарядов и их уроном
2) Класс танк. При создании экземпляра танка он ничем не заряжен 3) В методе main проходится по всем типов снарядов из энума, заряжает соответствующий заряд и производит выстрел

DenisGubernatorov committed:
KS-19-04-tank
1) Энум с типами зарядов и их уроном
2) Класс танк. При создании экземпляра танка он ничем не заряжен
3) В методе main проходится по всем типов снарядов из энума, заряжает соответствующий заряд и производит выстрел

ievetrovdev commented on src/main/kotlin/lesson19/lesson19_task4.kt:
```diff
@@ -1 +1,32 @@
+    fun getDamage(ammunition: Ammunition): Int = ammunition.damage
+}
+
+class Tank {
```
Предлагаю добавить поле с зарядом в основной конструктор и сделать его null по умолчанию, чтобы была опция создания с каким-либо снарядом, и уже внутри работать с этим свойством. Так код будет более гибким.

ievetrovdev commented on src/main/kotlin/lesson19/lesson19_task4.kt:
```diff
@@ -1 +1,32 @@
+    RED(20),
+    ;
+
+    fun getDamage(ammunition: Ammunition): Int = ammunition.damage
```
Хорошо, но этот метод вызывается у конкретного экземпляра и возвращает его свойство, поэтому логичнее и правильнее сделать так: `fun getDamage(): Int = this.damage`

DenisGubernatorov committed:
KS-19-04-tank
1) Рефакторинг lesson19.Ammunition.getDamage. В метод ничего не передается и работает с this
2) Рефакторинг класса Tank. Свойство damage переименовано в ammunitionType, вынесено в основной конструктор. Изменен тип. Теперь передается константа из Ammunition, а не количество дамага для константы.
3) Метод shoot доработан на работу с null. Если тип заряда не установлен, то выведется сообщение об осечке
4) В main добавлен тестовый объект Tank, для проверки кэйса, когда тип заряда null

ievetrovdev reviewed (approved):
Хорошо!
